webui/react/src/hooks/useModal/Model/useModalModelMove.tsx

```diff
@@ -0,0 +1,110 @@
+import { Select } from 'antd';
+import { ModalFuncProps } from 'antd/es/modal/Modal';
+import React from 'react';
+import { useCallback } from 'react';
+
+import Form from 'components/kit/Form';
+import Link from 'components/Link';
+import { WorkspaceDetailsTab } from 'pages/WorkspaceDetails';
+import { paths } from 'routes/utils';
+import { moveModel } from 'services/api';
+import useModal, { ModalHooks as Hooks, ModalCloseReason } from 'shared/hooks/useModal/useModal';
+import { useWorkspaces } from 'stores/workspaces';
+import { ModelItem } from 'types';
+import { notification } from 'utils/dialogApi';
+import handleError from 'utils/error';
+import { Loadable } from 'utils/loadable';
+
+type FormInputs = {
+  workspaceId: number;
+};
+
+interface Props {
+  onClose?: (reason?: ModalCloseReason) => void;
+}
+
+interface ModalHooks extends Omit<Hooks, 'modalOpen'> {
+  modalOpen: (model: ModelItem) => void;
+}
+
+const useModalModelMove = ({ onClose }: Props = {}): ModalHooks => {
+  const [form] = Form.useForm<FormInputs>();
+  const { modalOpen: openOrUpdate, ...modalHook } = useModal({ onClose });
+  const workspaces = Loadable.getOrElse([], useWorkspaces());
+
+  const getModalProps = useCallback(
+    (model: ModelItem): ModalFuncProps => {
+      const handleOk = async () => {
+        const values = await form.validateFields();
+        try {
+          await moveModel({ destinationWorkspaceId: values.workspaceId, modelName: model.name });
+          const workspaceName = workspaces.find((ws) => ws.id === values.workspaceId)?.name;
+          const path = paths.workspaceDetails(
+            values.workspaceId,
+            WorkspaceDetailsTab.ModelRegistry,
+          );
+          notification.success({
+            description: (
+              <div>
+                <p>
+                  {model.name} moved to workspace {workspaceName}
+                </p>
+                <Link path={path}>View Workspace</Link>
+              </div>
+            ),
+            key: 'move-model-notification',
+            message: 'Successfully Moved',
+          });
+        } catch (e) {
+          handleError(e, { publicSubject: `Unable to move model ${model.id}.`, silent: false });
+        } finally {
+          form.resetFields();
+        }
+      };
+
+      const handleClose = () => {
+        form.resetFields();
+        onClose?.();
+      };
+
+      return {
+        closable: true,
+        content: (
+          <Form autoComplete="off" form={form} layout="vertical">
+            <Form.Item
+              label="Workspace"
+              name="workspaceId"
+              rules={[{ message: 'Please select a workspace', required: true }]}>
+              <Select
+                filterOption={(input, option) => (option?.label ?? '').includes(input)}
+                filterSort={(a, b) => (a.label < b.label ? 1 : -1)}
+                options={workspaces.map((ws) => ({ label: ws.name, value: ws.id }))}
```

Comment:
Does it make sense to filter the `workspaces` here to only workspaces where a user has the appropriate `canMove` permissions.

Reply:
i think it does make sense. will update soon

Reply:
updated